File: samples/demos/irb/pkg/storage/py/storage_client.py
# Copyright 2021 Intel Corporation
#
# SPDX-License-Identifier: Apache-2.0

import redis
import logging
from os import environ

logger = logging.getLogger()

class StorageClient:
	def __init__(self):
	    self.host = environ.get('REDIS_HOST', 'localhost')
	    self.port = environ.get('REDIS_PORT', '6379')
	    self.db = '0'

	    self.r = redis.Redis(self.host, self.port, self.db)

	def set(self, key, value):
	    try:
	        self.r.set(key, value)
	    except Exception as e:
	        logger.error("Error set: {}".format(str(e)))
	        return False
	    return True

	def get(self, key):
	    try:
	        value = self.r.get(key)
	    except Exception as e:
	        logger.error("Error get: {}".format(str(e)))
	        return None, False
	    return value, True

File: samples/demos/irb/pkg/storage/go/storageclient.go
/*
   Copyright IBM Corp. All Rights Reserved.
   Copyright 2021 Intel Corporation

   SPDX-License-Identifier: Apache-2.0
*/

package storage

import (
	"context"
	"os"

	"github.com/go-redis/redis/v8"
)

var ctx = context.Background()

func getEnv(key, defaultValue string) string {
	value := os.Getenv(key)
	if value == "" {
		return defaultValue
	}
	return value
}

func newRedisClient() *redis.Client {
	var (
		host     = getEnv("REDIS_HOST", "localhost")
		port     = getEnv("REDIS_PORT", "6379")
		password = getEnv("REDIS_PASSWORD", "")
	)

	return redis.NewClient(&redis.Options{
		Addr:     host + ":" + port,
		Password: password, // no password set
		DB:       0,        // use default DB
	})
}

func Get(key []byte) (value []byte, e error) {
	rdb := newRedisClient()

	val, err := rdb.Get(ctx, string(key)).Bytes()
	if err != nil {
		return nil, err
	}

	return val, nil
}

func Set(key string, value string) (e error) {
	rdb := newRedisClient()

	err := rdb.Set(ctx, key, value, 0).Err()
	if err != nil {
		return err
	}
	return nil
}
